File: ruby/raindrops/raindrops.rb
class Raindrops
  def self.convert(number)
    raindrop_sound = ''

    (1..number).each do |factor|
      raindrop_sound += 'Pling' if number % factor == 0 and factor == 3
      raindrop_sound += 'Plang' if number % factor == 0 and factor == 5
      raindrop_sound += 'Plong' if number % factor == 0 and factor == 7
    end

    raindrop_sound.empty? ? number.to_s : raindrop_sound
  end
end

File: python/raindrops/raindrops.py
def convert(number):
    raindrop_sounds = ""

    if number % 3 == 0:
        raindrop_sounds += "Pling"
    if number % 5 == 0:
        raindrop_sounds += "Plang"
    if number % 7 == 0:
        raindrop_sounds += "Plong"
    if not raindrop_sounds:
        raindrop_sounds = str(number)

    return raindrop_sounds
